Crash when cloning a VM with existing EFI disk

```
  Enter a value: yes

proxmox_virtual_environment_vm.debian_vm: Creating...
proxmox_virtual_environment_vm.debian_vm: Still creating... [10s elapsed]
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v0.75.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xdc78c0]

goroutine 38 [running]:
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.vmCreateClone({0x1322748, 0xc00041c9d0}, 0xc000568400, {0x10635a0, 0xc000175880})
        /home/runner/work/terraform-provider-proxmox/terraform-provider-proxmox/proxmoxtf/resource/vm/vm.go:2238 +0x39a0
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.vmCreate({0x1322780, 0xc000230070}, 0xc000568400, {0x10635a0, 0xc000175880})
        /home/runner/work/terraform-provider-proxmox/terraform-provider-proxmox/proxmoxtf/resource/vm/vm.go:1610 +0xad
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000328d00, {0x13226d8, 0xc000423020}, 0xc000568400, {0x10635a0, 0xc000175880})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:838 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000328d00, {0x13226d8, 0xc000423020}, 0xc0000e28f0, 0xc0000e9480, {0x10635a0, 0xc000175880})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:969 +0xa8b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0001658d8, {0x13226d8?, 0xc000422ea0?}, 0xc000170a50)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:1188 +0xd1a
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x1330838?, 0xc0001658d8?}}, {0x13226d8, 0xc000422ea0}, 0x0?)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-mux@v0.18.0/tf5to6server/tf5to6server.go:38 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0xc0002ea680, {0x13226d8?, 0xc000422bd0?}, 0xc000170a00)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-mux@v0.18.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x18d
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0000c60a0, {0x13226d8?, 0xc000422090?}, 0xc0001740e0)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:866 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x10d7940, 0xc0000c60a0}, {0x13226d8, 0xc000422090}, 0xc0000e8a00, 0x0)
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:611 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000034200, {0x13226d8, 0xc0005cf800}, 0xc000247260, 0xc0000c1bf0, 0x1abae58, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.70.0/server.go:1400 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc000034200, {0x13232a8, 0xc0002f5860}, 0xc000247260)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.70.0/server.go:1810 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.70.0/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 21
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.70.0/server.go:1041 +0x11d

Error: The terraform-provider-proxmox_v0.75.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```
```
terraform@terraformansible:~/opentofu$ ls -al
total 32
drwxr-xr-x  3 terraform terraform 4096 Apr  8 14:30 .
drwxr-xr-x 21 terraform terraform 4096 Apr  8 14:20 ..
-rw-r--r--  1 terraform terraform 1107 Apr  8 14:28 main.tf
-rw-r--r--  1 terraform terraform  373 Apr  8 14:30 provider.tf
drwxr-xr-x  3 terraform terraform 4096 Apr  8 14:29 .terraform
-rw-r--r--  1 terraform terraform 1404 Apr  8 14:29 .terraform.lock.hcl
-rw-r--r--  1 terraform terraform  181 Apr  8 14:30 terraform.tfstate
-rw-r--r--  1 terraform terraform 1032 Apr  8 14:25 vars.tf
terraform@terraformansible:~/opentofu$ cat main.tf && cat provider.tf && cat vars.tf
```
```hcl
resource "proxmox_virtual_environment_vm" "debian_vm" {
    name        = "bpg-ubuntu-2204"
    description = "Managed by opentofu"
    tags        = ["opentofu", "ubuntu2204"]
    node_name = var.target_node
    clone {
        vm_id = "8002"
    }

    cpu {
        cores = 2
        type = "host"
        numa = true
    }
    memory {
        dedicated = 2048
    }
    network_device {
        bridge = "vmbr0"
        model = "virtio"
    }

    efi_disk {
        datastore_id = "local-lvm"
        file_format = "raw"
        type    = "4m"
    }

    disk {
        datastore_id = "local-lvm"
        file_format = "raw"
        interface = "scsi0"
        size = "10"
    }

    operating_system {
        type = "l26"
    }
    machine = "q35"
    agent {
        enabled = false
    }

    initialization {
        ip_config {
            ipv4 {
                address = "x.x.x.x/24"
                gateway = "x.x.x.x"
            }
        }
        user_account {
            keys     = [var.ssh_key]
            password = "tofu"
            username = "debian"
        }
    }
}
terraform {
    required_providers {
        proxmox =  {
        source = "bpg/proxmox"
        version = "0.75.0"
        }
    }
}
provider "proxmox" {
    endpoint = var.pm_api_url
    api_token = var.pm_api_token
    insecure = true  # Warning TLS cert of proxmox webui must be valid (Let's Encrypt)
    ssh {
        agent    = true
        username = "root"
    }
}
variable  "pm_api_url" {
    default =  "https://x.x.x.x:8006/api2/json"
}
variable  "pm_api_token" {
    default =  "opentofu@pve!opentofu=token"
}
variable  "target_node" {
    default =  "proxmox"
}
variable  "ssh_key" {
    default =  "ssh-rsa <sshpubkey>"
}
```

--- a/proxmoxtf/resource/vm/vm.go
+++ b/proxmoxtf/resource/vm/vm.go
@@ -2235,9 +2235,13 @@ func vmCreateClone(ctx context.Context, d *schema.ResourceData, m interface{}) d
 			continue
 		}
 
-		if efiType != *currentDiskInfo.Type {
+		// the efitype is either "2m" or "4m"
+		// so it's kinda resizing, but with 2 fixed sizes it treated as a "type" by PVE
+		// default is "2m", hence this elaborated check.
+		if (currentDiskInfo.Type != nil && efiType != *currentDiskInfo.Type) ||
+			(currentDiskInfo.Type == nil && efiType != dvEFIDiskType) {
 			return diag.Errorf(
-				"resizing of efidisks is not supported.",
+				"changing the EFI disk type (4m / 2m) is not supported",
 			)
 		}
 

